include/s3select_functions.h

```diff
@@ -25,65 +25,91 @@ struct push_char
 };
 static push_char g_push_char;
 
-struct push_1dig
+struct push_2dig
 {
   void operator()(const char* a, const char* b, uint32_t* n) const
   {
-    *n = (static_cast<char>(*a) - 48);
+    *n = (static_cast<char>(*a) - 48) * 10 + (static_cast<char>(*(a+1)) - 48) ;
   }
 
 };
-static push_1dig g_push_1dig;
+static push_2dig g_push_2dig;
 
-struct push_2dig
+struct push_4dig
 {
   void operator()(const char* a, const char* b, uint32_t* n) const
   {
-    *n = (static_cast<char>(*a) - 48) * 10 + (static_cast<char>(*(a+1)) - 48) ;
+    *n = (static_cast<char>(*a) - 48) * 1000 + (static_cast<char>(*(a+1)) - 48) * 100 + (static_cast<char>(*(a+2)) - 48) * 10 + (static_cast<char>(*(a+3)) - 48);
   }
 
 };
-static push_2dig g_push_2dig;
+static push_4dig g_push_4dig;
 
-struct push_3dig
+struct push_1fdig
 {
-  void operator()(const char* a, const char* b, uint32_t* n) const
+  void operator()(const char* a, const char* b, double* n) const
```

Comment:
no need for floating-point, it's less accurate.
the nano-sec should be uint32 (amount of nano-sec, not a fraction)
the push_Xfdig::operator() should multiply the result by 10^(6- number of digits) i.e. 1,10,100,...